# Question 1

In [1]:
library(tidyverse)

Warning message in system("timedatectl", intern = TRUE):
“running command 'timedatectl' had status 1”
── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──

✔ ggplot2 3.3.5     ✔ purrr   0.3.4
✔ tibble  3.1.7     ✔ dplyr   1.0.9
✔ tidyr   1.2.0     ✔ stringr 1.4.0
✔ readr   1.4.0     ✔ forcats 0.5.1

── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()



In [2]:
sneaker_df <- read_csv("sneaker_data.csv")
sneaker_df


── Column specification ────────────────────────────────────────────────────────
cols(
  order_id = col_double(),
  shop_id = col_double(),
  user_id = col_double(),
  order_amount = col_double(),
  total_items = col_double(),
  payment_method = col_character(),
  created_at = col_character()
)



order_id shop_id user_id order_amount total_items payment_method
1     1        53     746        224          2        cash          
2     2        92     925         90          1        cash          
3     3        44     861        144          1        cash          
4     4        18     935        156          1        credit_card   
5     5        18     883        156          1        credit_card   
6     6        58     882        138          1        credit_card   
7     7        87     915        149          1        cash          
8     8        22     761        292          2        cash          
9     9        64     914        266          2        debit         
10   10        52     788        146          1        credit_card   
11   11        66     848        322          2        credit_card   
12   12        40     983        322          2        debit         
13   13        54     799        266          2        credit_card   
14   14       100     709        111          1        cash          
15   15        87     849        447          3        credit_card   
16   16        42     607     704000       2000        credit_card   
17   17        17     731        176          1        cash          
18   18        28     752        164          1        credit_card   
19   19        83     761        258          2        cash          
20   20        63     898        408          3        credit_card   
21   21        66     987        322          2        cash          
22   22        97     789        486          3        credit_card   
23   23        88     985        704          4        credit_card   
24   24        75     964        256          2        credit_card   
25   25        73     917        495          3        cash          
26   26        82     848        177          1        cash          
27   27        47     882        145          1        cash          
28   28        53     942        112          1        credit_card   
29   29        40     944        322          2        cash          
30   30        59     790        178          1        credit_card   
⋮    ⋮        ⋮       ⋮       ⋮            ⋮           ⋮             
4971 4971      34     987     244          2           cash          
4972 4972      49     854     129          1           debit         
4973 4973      58     725     414          3           debit         
4974 4974      92     970     360          4           cash          
4975 4975       7     925     448          4           credit_card   
4976 4976      76     932     155          1           cash          
4977 4977      93     763     114          1           credit_card   
4978 4978      69     903     131          1           credit_card   
4979 4979       2     841     282          3           cash          
4980 4980      75     820     128          1           debit         
4981 4981      50     942     772          4           credit_card   
4982 4982      86     715     260          2           debit         
4983 4983       1     906     316          2           credit_card   
4984 4984      46     969     332          2           cash          
4985 4985      44     966     432          3           credit_card   
4986 4986       8     916     396          3           debit         
4987 4987     100     731     111          1           cash          
4988 4988      11     826     184          1           credit_card   
4989 4989      86     877     260          2           cash          
4990 4990      11     843     552          3           credit_card   
4991 4991      24     860     140          1           cash          
4992 4992      61     707     158          1           cash          
4993 4993      49     739     258          2           debit         
4994 4994       4     834     384          3           debit         
4995 4995      12     954     201          1           cash          
4996 4996      

In [3]:
head(sneaker_df)
dim(sneaker_df)
glimpse(sneaker_df)

order_id shop_id user_id order_amount total_items payment_method
1 1        53      746     224          2           cash          
2 2        92      925      90          1           cash          
3 3        44      861     144          1           cash          
4 4        18      935     156          1           credit_card   
5 5        18      883     156          1           credit_card   
6 6        58      882     138          1           credit_card   
  created_at         
1 2017-03-13 12:36:56
2 2017-03-03 17:38:52
3 2017-03-14 4:23:56 
4 2017-03-26 12:43:37
5 2017-03-01 4:35:11 
6 2017-03-14 15:25:01

[1] 5000    7

Rows: 5,000
Columns: 7
$ order_id       <dbl> 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, …
$ shop_id        <dbl> 53, 92, 44, 18, 18, 58, 87, 22, 64, 52, 66, 40, 54, 100…
$ user_id        <dbl> 746, 925, 861, 935, 883, 882, 915, 761, 914, 788, 848, …
$ order_amount   <dbl> 224, 90, 144, 156, 156, 138, 149, 292, 266, 146, 322, 3…
$ total_items    <dbl> 2, 1, 1, 1, 1, 1, 1, 2, 2, 1, 2, 2, 2, 1, 3, 2000, 1, 1…
$ payment_method <chr> "cash", "cash", "cash", "credit_card", "credit_card", "…
$ created_at     <chr> "2017-03-13 12:36:56", "2017-03-03 17:38:52", "2017-03-…


Let's check if there is any NA observations within our data. 

In [4]:
sum(is.na(sneaker_df))

[1] 0

Since we are interested in the average value of an order, we can check some of the statistics for the `order_amount` column.

In [5]:
summary(sneaker_df$order_amount)

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
     90     163     284    3145     390  704000 

In [6]:
sd(sneaker_df$order_amount)

[1] 41282.54

We can see that the minimum order amount was 90 dollars,  maximum order amount was 704,000 dollars, and therefore, the mean order amount in this data set is $3145.

We can analyze the data further by looking at the order values between different shops. In order to do that, we group the orders based on their `shop_id`'s and we calculate their mean and median order amounts and total items they sold. 

In [7]:
shop_analysis <- sneaker_df %>%
                group_by(shop_id) %>%
                summarize(num_of_orders = n(), 
                          mean_order_amount = mean(order_amount, na.rm = TRUE),
                          mean_total_items = mean(total_items, na.rm = TRUE),
                          median_order_amount = median(order_amount, na.rm = TRUE),
                          median_total_items = median(total_items, na.rm = TRUE),
                         )
shop_analysis

shop_id num_of_orders mean_order_amount mean_total_items
1    1      44            308.8182          1.954545        
2    2      55            174.3273          1.854545        
3    3      48            305.2500          2.062500        
4    4      51            258.5098          2.019608        
5    5      45            290.3111          2.044444        
6    6      59            383.5085          2.050847        
7    7      56            218.0000          1.946429        
8    8      46            241.0435          1.826087        
9    9      59            234.0000          1.983051        
10  10      53            332.3019          2.245283        
11  11      49            356.7347          1.938776        
12  12      53            352.6981          1.754717        
13  13      63            345.3968          2.158730        
14  14      58            242.0000          2.086207        
15  15      52            308.9423          2.019231        
16  16      41            270.1463          1.731707        
17  17      53            332.0755          1.886792        
18  18      51            342.5882          2.196078        
19  19      64            320.9062          1.968750        
20  20      52            251.5577          1.980769        
21  21      46            308.6957          2.173913        
22  22      48            273.7500          1.875000        
23  23      55            317.6727          2.036364        
24  24      55            320.7273          2.290909        
25  25      48            232.9167          1.791667        
26  26      49            341.2245          1.938776        
27  27      54            334.8704          1.981481        
28  28      43            320.3721          1.953488        
29  29      58            331.6207          2.034483        
30  30      56            295.0714          1.928571        
⋮   ⋮       ⋮             ⋮                 ⋮               
71   71     66              323.0303        1.969697        
72   72     46              309.5652        1.934783        
73   73     58              335.6897        2.034483        
74   74     38              306.0000        2.000000        
75   75     42              240.7619        1.880952        
76   76     42              321.0714        2.071429        
77   77     50              280.8000        1.800000        
78   78     46            49213.0435        1.913043        
79   79     54              328.4815        1.814815        
80   80     45              299.6667        2.066667        
81   81     59              384.0000        2.169492        
82   82     42              349.7857        1.976190        
83   83     42              248.7857        1.928571        
84   84     59              342.3051        2.237288        
85   85     35              329.2571        1.914286        
86   86     52              277.5000        2.134615        
87   87     52              292.2692        1.961538        
88   88     50              355.5200        2.020000        
89   89     61              379.1475        1.934426        
90   90     49              403.2245        2.265306        
91   91     54              325.9259        2.037037        
92   92     42              162.8571        1.809524        
93   93     59              214.4746        1.881356        
94   94     45              297.7778        2.222222        
95   95     39              318.7692        1.897436        
96   96     51              330.0000        2.156863        
97   97     48              324.0000        2.000000        
98   98     58              245.3621        1.844828        
99   99     54              339.4444        1.740741        
100 100     40              213.6750        1.925000        
    median_order_amount median_total_items
1   316                 2.0               
2   188                 2.0               
3   296                 2.0               
4   256                 2.0               
5   284

In [8]:
summary(shop_analysis$num_of_orders)

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  35.00   44.75   50.00   50.00   55.00   68.00 

By looking at the table above, we can see that most sneaker shops received around the same number of orders ranging between 35-68 with a mean and median of 50 orders per shop. However, sneaker shop with `shop_id` 78 has a lot higher average order amount (49213.04 dollars) compared to the order shops. So, we arrange the table to see if there are other shops which have higher `mean_order_amount`.   

In [9]:
shop_analysis %>%
            arrange(desc(mean_order_amount)) %>%
            head()

shop_id num_of_orders mean_order_amount mean_total_items median_order_amount
1 42      51            235101.4902       667.901961         704              
2 78      46             49213.0435         1.913043       51450              
3 50      44               403.5455         2.090909         386              
4 90      49               403.2245         2.265306         356              
5 38      35               390.8571         2.057143         380              
6 81      59               384.0000         2.169492         354              
  median_total_items
1 2                 
2 2                 
3 2                 
4 2                 
5 2                 
6 2

We can see that shops 42 and 78 have a lot higher `mean_order_amount` than the others, and if we look more carefully, we can see that the mean of the total items sold given in `mean_total_items` column is roughly 668 items for shop 42.

In [10]:
df_42 <- sneaker_df %>%
        filter(shop_id == 42) %>%
        arrange(desc(order_amount))
df_42

order_id shop_id user_id order_amount total_items payment_method
1    16     42      607     704000       2000        credit_card   
2    61     42      607     704000       2000        credit_card   
3   521     42      607     704000       2000        credit_card   
4  1105     42      607     704000       2000        credit_card   
5  1363     42      607     704000       2000        credit_card   
6  1437     42      607     704000       2000        credit_card   
7  1563     42      607     704000       2000        credit_card   
8  1603     42      607     704000       2000        credit_card   
9  2154     42      607     704000       2000        credit_card   
10 2298     42      607     704000       2000        credit_card   
11 2836     42      607     704000       2000        credit_card   
12 2970     42      607     704000       2000        credit_card   
13 3333     42      607     704000       2000        credit_card   
14 4057     42      607     704000       2000        credit_card   
15 4647     42      607     704000       2000        credit_card   
16 4869     42      607     704000       2000        credit_card   
17 4883     42      607     704000       2000        credit_card   
18 1365     42      797       1760          5        cash          
19 1368     42      926       1408          4        cash          
20 1472     42      907       1408          4        debit         
21  939     42      808       1056          3        credit_card   
22 2988     42      819       1056          3        cash          
23 3514     42      726       1056          3        debit         
24  410     42      904        704          2        credit_card   
25  836     42      819        704          2        cash          
26 1521     42      756        704          2        debit         
27 1912     42      739        704          2        cash          
28 2004     42      934        704          2        cash          
29 2274     42      747        704          2        debit         
30 2492     42      868        704          2        debit         
31 2610     42      868        704          2        debit         
32 2767     42      970        704          2        credit_card   
33 4295     42      859        704          2        cash          
34 4327     42      788        704          2        debit         
35 4422     42      736        704          2        credit_card   
36 4768     42      720        704          2        credit_card   
37   41     42      793        352          1        credit_card   
38  309     42      770        352          1        credit_card   
39  835     42      792        352          1        cash          
40  980     42      744        352          1        debit         
41 1513     42      946        352          1        debit         
42 1930     42      770        352          1        credit_card   
43 2019     42      739        352          1        debit         
44 2054     42      951        352          1        debit         
45 3652     42      830        352          1        credit_card   
46 3698     42      839        352          1        debit         
47 3904     42      975        352          1        debit         
48 3999     42      886        352          1        debit         
49 4232     42      962        352          1        cash          
50 4626     42      809        352          1        credit_card   
51 4746     42      872        352          1        debit         
   created_at         
1  2017-03-07 4:00:00 
2  2017-03-04 4:00:00 
3  2017-03-02 4:00:00 
4  2017-03-24 4:00:00 
5  2017-03-15 4:00:00 
6  2017-03-11 4:00:00 
7  2017-03-19 4:00:00 
8  2017-03-17 4:00:00 
9  2017-03-12 4:00:00 
10 2017-03-07 4:00:00 
11 2017-03-28 4:00:00 
12 2017-03-28 4:00:00 
13 2017-03-24 4:00:00 
14 2017-03-28 4:00:00 
15 2017-03-02 4:00:00 
16 2017-03-22 4:00:00 
17 2017-03-25 4:00:00 
18 2017-03-10 6:28:21 
19 2017-03-13 2:38:34 
20 2017

For shop 42, there are 17 credit card orders of 2000 sneakers bought for 704,000 dollars in March 2017 with the exact same customer or user id. This means that shop 42 might have some sort of a deal with user 607 to purchase 2000 items for $704,000.  

In [11]:
df_78 <- sneaker_df %>%
        filter(shop_id == 78) %>%
        arrange(desc(order_amount))
df_78

order_id shop_id user_id order_amount total_items payment_method
1   692     78      878     154350       6           debit         
2  2493     78      834     102900       4           debit         
3  1260     78      775      77175       3           credit_card   
4  2565     78      915      77175       3           debit         
5  2691     78      962      77175       3           debit         
6  2907     78      817      77175       3           debit         
7  3404     78      928      77175       3           debit         
8  3725     78      766      77175       3           credit_card   
9  4193     78      787      77175       3           credit_card   
10 4421     78      969      77175       3           debit         
11 4716     78      818      77175       3           debit         
12  491     78      936      51450       2           debit         
13  494     78      983      51450       2           cash          
14  512     78      967      51450       2           cash          
15  618     78      760      51450       2           cash          
16 1530     78      810      51450       2           cash          
17 2453     78      709      51450       2           cash          
18 2496     78      707      51450       2           cash          
19 2513     78      935      51450       2           debit         
20 2819     78      869      51450       2           debit         
21 2822     78      814      51450       2           cash          
22 3102     78      855      51450       2           credit_card   
23 3168     78      927      51450       2           cash          
24 3706     78      828      51450       2           credit_card   
25 4080     78      946      51450       2           cash          
26 4312     78      960      51450       2           debit         
27 4413     78      756      51450       2           debit         
28  161     78      990      25725       1           credit_card   
29 1057     78      800      25725       1           debit         
30 1194     78      944      25725       1           debit         
31 1205     78      970      25725       1           credit_card   
32 1385     78      867      25725       1           cash          
33 1420     78      912      25725       1           cash          
34 1453     78      812      25725       1           credit_card   
35 2271     78      855      25725       1           credit_card   
36 2549     78      861      25725       1           cash          
37 2774     78      890      25725       1           cash          
38 2923     78      740      25725       1           debit         
39 3086     78      910      25725       1           cash          
40 3152     78      745      25725       1           credit_card   
41 3441     78      982      25725       1           debit         
42 3781     78      889      25725       1           cash          
43 4041     78      852      25725       1           cash          
44 4506     78      866      25725       1           debit         
45 4585     78      997      25725       1           cash          
46 4919     78      823      25725       1           cash          
   created_at         
1  2017-03-27 22:51:43
2  2017-03-04 4:37:34 
3  2017-03-27 9:27:20 
4  2017-03-25 1:19:35 
5  2017-03-22 7:33:25 
6  2017-03-16 3:45:46 
7  2017-03-16 9:45:05 
8  2017-03-16 14:13:26
9  2017-03-18 9:25:32 
10 2017-03-09 15:21:35
11 2017-03-05 5:10:44 
12 2017-03-26 17:08:19
13 2017-03-16 21:39:35
14 2017-03-09 7:23:14 
15 2017-03-18 11:18:42
16 2017-03-29 7:12:01 
17 2017-03-27 11:04:04
18 2017-03-26 4:38:52 
19 2017-03-18 18:57:13
20 2017-03-17 6:25:51 
21 2017-03-02 17:13:25
22 2017-03-21 5:10:34 
23 2017-03-12 12:23:08
24 2017-03-14 20:43:15
25 2017-03-20 21:14:00
26 2017-03-01 3:02:10 
27 2017-03-02 4:13:39 
28 2017-03-12 5:56:57 
29 2017-03-15 10:16:45
30 2017-03-16 16:38:26
31 2017-03-17 22:32:21
32 2017-03-17 16:38:06
33 2017-03-30 12:23:43
34 2017-03-17 18:09:54
35

On the other hand, shop 78 had orders worth 154,350 and 102,900 dollars, but the last observations above tell us that shop 78 sold 1 item for 25,725 dollars to different customers. This suggests that shop 78 could be selling more luxury or higher end products than the other shops. 

These expensive orders placed on shops 42 and 78 cause our analysis of average order value (AOV) to be skewed, and result in AOV of $3145.15. 

### Alternate Option

Because of the fact that there are orders purchased in bulk or maybe higher end or rare sneakers sold by some shops, we get the average order value (AOV) to be skewed by these transactions. Therefore, a good way for us to evaluate this data is to use the median or the mean of the order value per item. 

In order to do that, we need to create a new column called `order_val_per_item` and for each order id, we need to divide the `order_amount` by `total_items`.

In [12]:
mutated_df <- sneaker_df %>%
            mutate(order_val_per_item = order_amount/total_items)
mutated_df %>% head() # look at the head 
dim(mutated_df) # should be 5000x8

order_id shop_id user_id order_amount total_items payment_method
1 1        53      746     224          2           cash          
2 2        92      925      90          1           cash          
3 3        44      861     144          1           cash          
4 4        18      935     156          1           credit_card   
5 5        18      883     156          1           credit_card   
6 6        58      882     138          1           credit_card   
  created_at          order_val_per_item
1 2017-03-13 12:36:56 112               
2 2017-03-03 17:38:52  90               
3 2017-03-14 4:23:56  144               
4 2017-03-26 12:43:37 156               
5 2017-03-01 4:35:11  156               
6 2017-03-14 15:25:01 138

[1] 5000    8

We can analyze this further by using the summary function after we select the columns we need.  

In [13]:
df <- mutated_df %>% select(order_id, shop_id, user_id, order_val_per_item)
dim(df) # should be 5000x4
head(df)

[1] 5000    4

order_id shop_id user_id order_val_per_item
1 1        53      746     112               
2 2        92      925      90               
3 3        44      861     144               
4 4        18      935     156               
5 5        18      883     156               
6 6        58      882     138

In [14]:
summary(df$order_val_per_item)

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   90.0   133.0   153.0   387.7   169.0 25725.0 

We can see that the median of the order value per sneaker sold is 153 dollars, and the mean of the order value per sneaker sold is 387.7 dollars. This shows that the shop_id= 78 still affects our mean value quite a bit since it only sells one model of shoe from 25,725 dollars shown below.

In [15]:
mutated_df_analysis <- mutated_df %>%
                group_by(shop_id) %>%
                summarize(num_of_orders = n(), 
                          mean_order_val_per_item = mean(order_val_per_item, na.rm = TRUE)
                         )
shop_78_analysis <- filter(mutated_df_analysis, shop_id == 78)
shop_78_analysis

shop_id num_of_orders mean_order_val_per_item
1 78      46            25725

In this case, we could remove the orders with `shop_id` = 78, and note this shop id to further investigate to see if this shop sells luxury sneakers or some other rare items. After removing the outlier, we can check the average order value per item. 

In [16]:
df_78_removed <- df %>%
        filter(shop_id != 78)
summary(df_78_removed$order_val_per_item)

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   90.0   132.0   153.0   152.5   168.0   352.0 

When we remove the observations or orders sold by shop id 78, we get the mean of our order value per item as 152.5 dollars which is almost the same as the median of our order value per item (153 dollars).

### Conclusion

Therefore, we can conclude that the best way to evaluate this data to is to look at the `median of average order value per item`. This gives us a lot better idea about the prices of sneakers sold on Shopify, because median isn't much affected by the extreme cases within our data such as the 17 orders of 2000 sneakers or the sneakers sold by `shop_id` 78.

The value for this `median of average order value per item` metric used in this data set is $153.